src/pipeline_migration/actions/add_task.py

```diff
@@ -0,0 +1,406 @@
+from pipeline_migration.quay import get_active_tag
+from pipeline_migration.registry import REGISTRY, Container
+from pipeline_migration.types import FilePath
+from pipeline_migration.utils import load_yaml, dump_yaml, YAMLStyle
+
+logger = logging.getLogger("add_task")
+
+
+SUBCMD_DESCRIPTION: Final = """\
+The following are several examples with a Konflux task push-dockerfile:
+
+* Add task with latest bundle to pipelines within relative .tekton/ directory.
+
+    cd /path/to/repo
+    pipeline-migration-tool add-task push-dockerfile
+
+* Add task to mulitple pipelines in several repositories:
+
+    pipeline-migration-tool add-task push-dockerfile \\
+        /path/to/repo1/.tekton/pr.yaml /path/to/repo2/.tekton/push.yaml
+
+* Add task with parameter and execution order:
+
+    pipeline-migration-tool add-task push-dockerfile \\
+        --param param1,value1 --param param2,value2 \\
+        --run-after build-image-index
+
+* Add task with specific bundle reference:
+
+    pipeline-migration-tool add-task --bundle-ref <bundle-reference> push-dockerfile
+"""
+
+
+class InconsistentBundleBuild(Exception):
+    """Registry does not have expected bundle image"""
+
+
+class KonfluxTaskNotExist(Exception):
+    """Konflux task is not found from build-definitions"""
+
+
+class KonfluxTaskFileNotExist(Exception):
+    """Konflux task file does not exist in a version-specific task directory"""
+
+
+def konflux_task_bundle_reference(value: str) -> str:
+    """Argument type for checking input bundle reference
+
+    :raises argparse.ArgumentTypeError: if input bundle reference is invalid.
+    """
+    build_def = KonfluxBuildDefinitions()
+    try:
+        build_def.validate_bundle_ref(value)
+    except ValueError as e:
+        raise ArgumentTypeError(str(e))
+    return value
+
+
+def task_param(value: str) -> tuple[str, str]:
+    parts = value.split(",", 1)
+    if len(parts) == 1:
+        raise ArgumentTypeError("Missing parameter name or value.")
+    return parts[0], parts[1]
+
+
+def register_cli(subparser) -> None:
+    add_task_parser = subparser.add_parser(
+        "add-task",
+        help="Add a Konflux task to build pipelines.",
+        description=SUBCMD_DESCRIPTION,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    add_task_parser.add_argument(
+        "task",
+        help="Konflux task name to add. This is the actual task name defined in "
+        "konflux-ci/build-definitions. If a trusted artifact task is being added, "
+        "ensure the task name has suffix -oci-ta, e.g. buildah-oci-ta.",
+    )
+    add_task_parser.add_argument(
+        "file_or_dir",
+        nargs="*",
+        help="Specify pipeline files to where the task is added. "
+        "A pipeline can be included in a PipelineRun definition or a single Pipeline itself. "
+        "Directory path can also be specified alternatively. %(prog)s searches pipelines "
+        "from all YAML files from the first level of the directory.",
```

Comment:
"All YAML files from the first level of the directory" feels too ambiguous, it could mean only the YAML files in the directory root.

What about "all the YAML files at most one subdirectory deep"?

Reply:
On second thought, maybe the current behavior is fine if we just document that the default path is `./.tekton`

Reply:
The help is missing some information, I'll add them. The current behaviors are:

* If `file_or_dir` is omitted, search YAML files from `./.tekton/`.
* If `file_or_dir` is a file, use it.
* If `file_or_dir` is a directory, search YAML files from the given directory.

For given directories, only search YAML files at the first level, no recursion.